compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/SwitchGenerator.kt

```diff
@@ -348,7 +348,27 @@ class SwitchGenerator(private val expression: IrWhen, private val data: BlockInf
         override fun shouldOptimize() = cases.size > 1
 
         override fun genSwitch() {
-            subject.accept(codegen, data).materialize()
+            // Do not generate line numbers for the table switching. In particular,
+            // the subject is extracted from the condition of the first branch which
+            // will give the wrong stepping behavior for code such as:
+            //
+            // when {
+            //   x == 42 -> 1
+            //   x == 32 -> 2
+            //   x == 24 -> 3
+            //   ...
+            // }
+            //
+            // If the subject line number is generated, we will not stop on the line
```

Comment:
Would it be line numbers for initialzier of when variable (`when (val x = smthTrickyCall()){...} or nested multiline  when(x = when {... } ) {...}`)? 
For example:
```   
   val y = 1;
    when (val x =
            when {
                y == 0 -> 1;
                y == 1 -> 2;
                else -> 0;
            }) {
        0 -> 123;
        1 -> {
        }
    }
```

Reply:
`subject` in both `IntSwitch` and `StringSwitch` has to be an `IrGetValue`, so this code does not come into play for such more complicated subjects. I added your test case for good measure.